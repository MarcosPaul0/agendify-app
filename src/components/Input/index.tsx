import { View, Text, TextInput } from 'react-native';
import { Controller } from 'react-hook-form';
import { COLORS } from '@constants/colors.constant';
import type { IControlledInputProps } from './interfaces/controlledInputProps.interface';

export function ControlledInput({
  label,
  errorMessage,
  inputProps,
  controllerProps,
  containerStyle = 'w-full',
}: IControlledInputProps) {
  const inputBorderStyle = errorMessage ? 'border-ERROR' : 'border-GRAY_500';

  return (
    <View className={containerStyle}>
      <Text className="text-base text-GRAY_800 font-bold">{label}</Text>
      <Controller
        {...controllerProps}
        render={({ field: { value, onChange } }) => (
          <TextInput
            {...inputProps}
            className={`
              text-base text-GRAY_800 w-full py-1.5 pl-2.5
              bg-GRAY_100 border ${inputBorderStyle} rounded-lg
            `}
            selectionColor={COLORS.GRAY_800}
            value={value}
            onChangeText={onChange}
          />
        )}
      />
      <Text className="text-ERROR pl-1 h-6">{errorMessage}</Text>
    </View>
  );
}
