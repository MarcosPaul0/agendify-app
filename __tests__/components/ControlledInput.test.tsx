import { render } from '@testing-library/react-native';
import { ControlledInput } from '@components/Input';
import { useForm } from 'react-hook-form';

describe('Controlled Input', () => {
  it('Match with snapshot', async () => {
    function ControlledInputComponent() {
      const { control } = useForm();

      return (
        <ControlledInput controllerProps={{ name: 'fake-name', control }} />
      );
    }

    const tree = render(<ControlledInputComponent />);

    expect(tree).toMatchSnapshot();
  });
});
