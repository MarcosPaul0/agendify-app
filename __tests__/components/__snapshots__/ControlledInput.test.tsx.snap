// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controlled Input Match with snapshot 1`] = `
<View
  style={
    [
      {
        "width": "100%",
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "fontSize": 16,
          "lineHeight": 24,
        },
        {
          "color": "#343A40",
        },
        {
          "fontWeight": "700",
        },
      ]
    }
  />
  <TextInput
    onChangeText={[Function]}
    selectionColor="#343A40"
    style={
      [
        {
          "fontSize": 16,
          "lineHeight": 24,
        },
        {
          "color": "#343A40",
        },
        {
          "width": "100%",
        },
        {
          "paddingBottom": 6,
          "paddingTop": 6,
        },
        {
          "paddingLeft": 10,
        },
        {
          "backgroundColor": "#F5F5F5",
        },
        {
          "borderBottomWidth": 1,
          "borderLeftWidth": 1,
          "borderRightWidth": 1,
          "borderTopWidth": 1,
        },
        {
          "borderBottomColor": "#ADB5BD",
          "borderLeftColor": "#ADB5BD",
          "borderRightColor": "#ADB5BD",
          "borderTopColor": "#ADB5BD",
        },
        {
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
        },
        {
          "maxHeight": 80,
        },
      ]
    }
    testID="fake-name"
  />
  <Text
    style={
      [
        {
          "color": "#EF4444",
        },
        {
          "paddingLeft": 4,
        },
        {
          "height": 24,
        },
      ]
    }
  />
</View>
`;
